src/pyspiffe/config.py

```diff
@@ -0,0 +1,90 @@
+import os
+import ipaddress
+from urllib.parse import ParseResult, urlparse
+from typing import Dict, List, Tuple
+
+
+SPIFFE_ENDPOINT_SOCKET = 'SPIFFE_ENDPOINT_SOCKET'
+
+
+class ConfigSetter:
+    def __init__(self):
+        self._apply_default_config()
+        self._apply_environment_variables()
+
+    def get_config(self) -> Dict:
+        return self._config
+
+    def _apply_default_config(self):
```

Comment:
nit: let's be explicit on the return type `-> None` in all pleases please. That helps to be consistent with other modules.